conda/lock.py

```diff
@@ -111,11 +114,15 @@ def __init__(self, directory_path, retries=10):
         self.lock_file_path = "%s.pid{0}.%s" % (lock_path_pre, LOCK_EXTENSION)
         # e.g. if locking directory `/conda`, lock file will be `/conda/conda.pidXXXX.conda_lock`
         self.lock_file_glob_str = "%s.pid*.%s" % (lock_path_pre, LOCK_EXTENSION)
+        # make sure '/' exists
         assert isdir(dirname(self.directory_path)), "{0} doesn't exist".format(self.directory_path)
         if not isdir(self.directory_path):
-            os.makedirs(self.directory_path, exist_ok=True)
-            log.debug("forced to create %s", self.directory_path)
-        assert os.access(self.directory_path, os.W_OK), "%s not writable" % self.directory_path
```

Comment:
I think we should always use try/except block when make directories or touch files. Moreover, sometime, we may not able to write, and I don't think we need assertion in this case

Moreover, my new added test catch one error, exist_ok=True not working on python27